signac/contrib/project.py

```diff
@@ -135,36 +115,29 @@ class Project:
     def __init__(self, config=None, _ignore_schema_version=False):
         if config is None:
             config = load_config()
-        self._config = _ProjectConfig(
-            config, _mutate_hook=partial(_invalidate_config_cache, self)
-        )
+        self._config = _ProjectConfig(config)
         self._lock = RLock()
 
-        # Prepare cached properties derived from the project config.
-        self._id = None
-        self._rd = None
-        self._wd = None
-
         # Ensure that the project id is configured.
-        if self.id is None:
+        try:
+            self._id = str(self.config["project"])
+        except KeyError:
```

Comment:
Any chance you could add a quick test for this? Should be fairly straight-forward.

Reply:
Done in 994ab740ab5397f9be79e11f9d2de3a91541578d.